dao/source_dao.go

```diff
@@ -13,12 +13,17 @@ import (
 
 // GetSourceDao is a function definition that can be replaced in runtime in case some other DAO provider is
 // needed.
-var GetSourceDao func(*int64) SourceDao
+var GetSourceDao func(*SourceDaoParams) SourceDao
 
 // getDefaultRhcConnectionDao gets the default DAO implementation which will have the given tenant ID.
-func getDefaultSourceDao(tenantId *int64) SourceDao {
+func getDefaultSourceDao(daoParams *SourceDaoParams) SourceDao {
+	var tenantID *int64
+	if daoParams != nil {
```

Comment:
I am not sure, it I understand correctly what is expected here, but there should be `if daoParams.TenantID != nil` or?

Reply:
The first check is required since `daoParams` is a pointer to `SourceDaoParams`. If you do `daoParams.TenantID != nil` directly you may end up with a pointer dereference error.

I guess the intent of Libor on this one is to pass a `nil` tenant ID if it is missing, and that way the queries won't return anything meaningful. But I don't know.

Reply:
thanks Mikel, it makes sense to me now

Reply:
yes as @MikelAlejoBR said but it is right when we call it with empty struct `&dao.SourceDaoParams{})` we can have dereference error - so I will fix it.